lib/Runtime/PlatformAgnostic/Platform/Windows/ArraysUtility.cpp

```diff
@@ -31,16 +31,6 @@ namespace Arrays
             AssertMsg(FALSE, "GetLocaleInfo failed");
             return false;
         }
-        else
-        {
-            // Append ' ' if necessary
-            if (*sepOutSize < 2 || szSeparator[*sepOutSize - 2] != ' ')
```

Comment:
Hint to unlock the Windows test failure: what value would `*sepOutSize` have when separator is just a string containing a single character and would that work for what it is used for in `JavaScriptArray.cpp`? (see my other comment)  `sepOutSize` comes out of a system function (`GetLocaleInfoEx`) on line 27 above; MSDN has a documentation page on it.

Reply:
if i understand correctly, seems like we using the GetLocaleInfoEx to get the size of the locale separator, then we use that size to copy over from scriptcontext to our szSeparator buffer?

from msdn, since cchdata that is passed into the GetLocaleInfoEx is unsigned 32bit int value of 6, so sepOutSize will have value of 2?
looks like need to append that null terminating character to the szSeparator in the else statement, if not it wonr even go into that JavaScriptArray.cpp if scope

Reply:
Yes, you are right - the system call on Windows counts the null character, while the consuming code in JavaScriptArray (and  other places) does not. We just need to make the size here match what the consuming code expects. Right now it seem to copy null character in JavaScriptArray. It might be a good idea to add a comment that we expect to not count null character.

Reply:
Oh I see, thanks! So can I just add the else statement again and add the null character when sepOutSize is 1 now like below?

else{
            if (*sepOutSize < 2)
            {
                szSeparator[*sepOutSize] = '\0';
            }
}

Reply:
I think you need to decrement `*sepOutSize`, regardless of what the size is, since syscall returns value too high by one:

```
+        else
+        {
+            // We need just the number of regular characters, but Win32 API counts terminating null character as well
+            --(*sepOutSize);
+        }
+
```

Reply:
Thank for the help! 😊 Where to go from here? do I close this? or wait for it to get merged

Reply:
It would get closed when the change gets merged, and then I'll check if the issue gets closed too :)